### PR TITLE
Arwen versioning mst

### PR DESCRIPTION
**Pull request by sasurobert**
Arwen versioning to master directly

### CODE REVIEW

File: process/smartContract/process.go [hunk 1]
```diff
@@ -1329,14 +1338,17 @@ func (sc *scProcessor) DeploySmartContract(tx data.TransactionHandler, acntSnd s
...
 	vmExec, err := sc.vmContainer.Get(vmType)
 	if err != nil {
+		sc.arwenChangeLocker.RUnlock()
 		log.Trace("VM not found", "error", err.Error())
 		return vmcommon.UserError, sc.ProcessIfError(acntSnd, txHash, tx, err.Error(), []byte(""), snapshot, vmInput.GasLocked)
 	}
 
 	vmOutput, err = vmExec.RunSmartContractCreate(vmInput)
 	if err != nil {
+		sc.arwenChangeLocker.RUnlock()
```
Comment (camilbancioiu) [inline process/smartContract/process.go:1351]: This should be above the `if`.

File: genesis/process/shardGenesisBlockCreator.go [hunk 2]
```diff
@@ -292,7 +296,8 @@ func createProcessorsForShardGenesisBlock(arg ArgsGenesisBlockCreator, generalCo
 		DeployEnableEpoch:              arg.GeneralConfig.SCDeployEnableEpoch,
 		AheadOfTimeGasUsageEnableEpoch: arg.GeneralConfig.AheadOfTimeGasUsageEnableEpoch,
 		ArwenV3EnableEpoch:             arg.GeneralConfig.RepairCallbackEnableEpoch,
-		ArwenESDTFunctionsEnableEpoch:  arg.GeneralConfig.ArwenESDTFunctionsEnableEpoch,
+		EpochNotifier:                  epochNotifier,
+		ArwenChangeLocker:              &sync.RWMutex{},
```
Comment (iulianpascalau) [inline genesis/process/shardGenesisBlockCreator.go:300]: should be the same pointer with the one on the L394 (on `development` branch I have used the same pointer in these 2 locations)

File: process/smartContract/process.go (hunk 1 above)
Comment (sasurobert) [reply process/smartContract/process.go:1351]: yes.

File: integrationTests/testProcessorNode.go [hunk 3]
```diff
@@ -1227,6 +1233,7 @@ func (tpn *TestProcessorNode) initResolvers() {
 }
 
 func (tpn *TestProcessorNode) initInnerProcessors(gasMap map[string]map[string]uint64) {
+	tpn.ArwenChangeLocker = &sync.RWMutex{}
```
Comment (iulianpascalau) [inline integrationTests/testProcessorNode.go:1236]: we should set the ArwenChangeLocker whenever we instantiate the TestProcessorNode struct (this is how was done on the `development` branch)

File: process/smartContract/process.go (hunk 1 above)
Comment (iulianpascalau) [inline process/smartContract/process.go]: This should be moved on L1350. Same problem on the development-based branch :facepalm:

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

File: process/smartContract/process.go (hunk 1 above)
Comment (sasurobert) [reply process/smartContract/process.go]: moved it.

File: integrationTests/testProcessorNode.go (hunk 3 above)
Comment (sasurobert) [reply integrationTests/testProcessorNode.go:1236]: this part is called by all tests - so this instantiation is correct. there will be a lot of conflicts with development. most probably the merge will be done like accept everything from dev.

File: genesis/process/shardGenesisBlockCreator.go (hunk 2 above)
Comment (sasurobert) [reply genesis/process/shardGenesisBlockCreator.go:300]: done

**Review by iulianpascalau — DISMISSED** [review]

File: integrationTests/testProcessorNode.go (hunk 3 above)
Comment (iulianpascalau) [reply integrationTests/testProcessorNode.go:1236]: Yup, exactly

**Review by camilbancioiu — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Review by gabi-vuls — APPROVED** [review]
System test passed

**Review by iulianpascalau — APPROVED** [review]

**Review by pislaProgramming — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**